### ISSUE TITLE
Empty spans raise an IndexError in 2.3.5

### ISSUE THREAD

**Issue by sam-qordoba** [explosion/spaCy#7972]
This is related to #6755, but I don't believe it is identical

## How to reproduce the behaviour

using spaCy 2.3.2:
```python
>>> len(doc)
6
>>> span = doc[6:]
>>> print(span)

```
prints an empty string.

Using 2.3.5:
```python
>>> len(doc)
6
>>> span = doc[6:]
>>> print(span)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "span.pyx", line 166, in spacy.tokens.span.Span.__repr__
  File "span.pyx", line 503, in spacy.tokens.span.Span.text.__get__
  File "span.pyx", line 190, in spacy.tokens.span.Span.__getitem__
IndexError: [E201] Span index out of range.
```

I am not saying this change in behavior is unreasonable, but it seems like it should be at least a minor version increment.

## Your Environment

* Operating System: macOS-10.16-x86_64-i386-64bit
* Python Version Used: 3.8.5
* spaCy Version Used: 2.3.2 / 2.3.5
* Environment Information:

**Comment by adrianeboyd** [comment]
Hi, I think it's the same problem as in the linked bug report and we're in agreement that it's a bug and not the intended behavior.

The fix is already in the `v2.x` branch but with the focus v3, we haven't made a patch release for v2.3.x since v2.3.5.

**Comment by adrianeboyd** [comment]
Hi, this should be fixed in v2.3.6.

Unfortunately there's a delay uploading the release to PyPI because we've hit the default project size, but if you'd like to use it before then, the wheels are attached here: https://github.com/explosion/spaCy/releases/tag/v2.3.6
